source/Preprocessor.cpp

```diff
@@ -395,28 +396,70 @@ Trivia Preprocessor::handleEndIfDirective(Token directive) {
     return parseBranchDirective(directive, Token(), taken);
 }
 
-void Preprocessor::expectTimescaleSpecifier(Token& value, Token& unit) {
-    // TODO: check for allowed values
-    auto token = peek();
-    if (token.kind == TokenKind::IntegerLiteral) {
+bool Preprocessor::expectTimescaleSpecifier(Token& value, Token& unit) {
+    if (peek().kind == TokenKind::IntegerLiteral) {
         value = consume();
         unit = expect(TokenKind::Identifier);
+    } else {
+        auto token = expect(TokenKind::TimeLiteral);
+        if (token.isMissing()) {
+            value = Token::createMissing(alloc, TokenKind::IntegerLiteral, token.location());
+            unit = Token::createMissing(alloc, TokenKind::Identifier, token.location());
+            return false;
+        }
+
+        const SVInt *val = std::get_if<SVInt>(&token.numericValue());
```

Comment:
Even with peek, this check is necessary to make sure it isn't a double, as then it would be a valid TimeLiteral but not valid for a Timescale specifier